xbmc/filesystem/AddonsDirectory.cpp

```diff
@@ -118,6 +133,37 @@ static void SetUpdateAvailProperties(CFileItemList &items)
   }
 }
 
+
+// Creates categories for addon types, if we have any addons with that type.
+static void GenerateTypeListing(
+  const CURL& path,
+  const std::vector<TYPE>& types,
+  const VECADDONS& addons,
+  CFileItemList& items)
+{
+  for (const auto& type : types)
```

Comment:
Wouldn't it be more efficient to loop over all addons and use `IsLookAndFeelType()` on every addon / addon's type?  That way you save an explicit loop. Furthermore you always have to loop over all addons but you don't necessarily have to loop over all types.